### PR TITLE
Added Query builder tests and refactoring

### PR DESCRIPTION
**Pull request by HarrisonHough**
## Description

-   Added test cases for query builder and some minor refactoring

#### Added

-   test cases for query builder

#### Updated

-   minor refactoring of config processor and query builder

## How to Test

-   Add steps to locally test these changes

## Checklist

-   [ ] Tests written or updated for the changes.
-   [ ] Documentation is updated.
-   [ ] Changelog is updated.
-   [ ] QA Testing is updated.

## QA Testing

-   Mention any useful test cases for this feature and add to QA Testing documentation.

## Details

-   If more details are necessary to support the description with images and videos add them here.

### CODE REVIEW

File: Runtime/Utils/AvatarAPIParameters.cs [hunk 1]
```diff
@@ -14,5 +14,6 @@ public static class AvatarAPIParameters
         public const string USE_MESHOPT = "useMeshOptCompression";
         public const string RENDER_BLEND_SHAPES = "blendShapes";
         public const string RENDER_SCENE = "scene";
+        public const string QUALITY = "quality";
     }
 }
```
Comment (srcnalt) [inline Runtime/Utils/AvatarAPIParameters.cs:19]: missing newline

File: Tests/Editor/QueryBuilderTests.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            queryBuilder.AddKeyValue(AvatarAPIParameters.QUALITY,"low" );
+            queryBuilder.AddKeyValue(AvatarAPIParameters.MESH_LOD,0.ToString() );
+            Assert.AreEqual(queryBuilder.Query, QUALITY_LOW_MESH_LOD);
+        }
+        
+        [Test]
+        public void Texture_Atlas_1024_MorphTargets()
+        {
+            var queryBuilder = new QueryBuilder();
+            queryBuilder.AddKeyValue(AvatarAPIParameters.TEXTURE_ATLAS,512.ToString() );
```
Comment (srcnalt) [inline Tests/Editor/QueryBuilderTests.cs:25]:
```suggestion
            queryBuilder.AddKeyValue(AvatarAPIParameters.TEXTURE_ATLAS, 512.ToString());
```

Comment (srcnalt) [reply Tests/Editor/QueryBuilderTests.cs:25]: why not directly add these numbers are strings, any specific reason?

**Review by srcnalt — COMMENTED** [review]
Comments and stuff up there.

File: Tests/Editor/HelperTests.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+using NUnit.Framework;
+using UnityEngine;
+
+namespace ReadyPlayerMe.AvatarLoader.Tests
```
[the remark this answers is not in the document]
Comment (HarrisonHough) [reply Tests/Editor/HelperTests.cs:4]: yep well my thought here was that if eventually we make some "extras" module then we could move the tests to there.

File: Tests/Editor/QueryBuilderTests.cs (hunk 2 above)
Comment (HarrisonHough) [reply Tests/Editor/QueryBuilderTests.cs:25]: yes good point :D

Comment (HarrisonHough) [reply Tests/Editor/QueryBuilderTests.cs]: fixed

File: Runtime/Utils/AvatarAPIParameters.cs (hunk 1 above)
Comment (HarrisonHough) [reply Runtime/Utils/AvatarAPIParameters.cs]: fixed

**Review by srcnalt — APPROVED** [review]

File: Tests/Editor/QueryBuilderTests.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            queryBuilder.AddKeyValue(AvatarAPIParameters.QUALITY, "low");
+            queryBuilder.AddKeyValue(AvatarAPIParameters.MESH_LOD,"0");
+            Assert.AreEqual(queryBuilder.Query, QUALITY_LOW_MESH_LOD);
+        }
+        
+        [Test]
+        public void Texture_Atlas_1024_MorphTargets()
+        {
+            var queryBuilder = new QueryBuilder();
+            queryBuilder.AddKeyValue(AvatarAPIParameters.TEXTURE_ATLAS,"1024" );
```
[the remark this answers is not in the document]
Comment (HarrisonHough) [reply Tests/Editor/QueryBuilderTests.cs]: fixed

**Pull request merged**